ember/app/components/project-detailed/component.js

```diff
@@ -0,0 +1,36 @@
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+export default class ProjectDetailedComponent extends Component {
+  @service notification;
+  @service store;
+  @service fetch;
+  @tracked loading = false;
+
+  @action
+  async syncProject() {
+    try {
+      this.loading = true;
+      // post request to the api endpoint to sync the project
+      const request = await this.fetch.fetch(
+        `/api/projects/${this.args.project.id}/sync`,
+        {
+          method: 'POST',
+        }
+      );
+      if (request.ok) {
+        this.notification.success('Project synced successfully');
+      } else if (request.status === 404) {
+        this.notification.danger('Project not found');
+      } else if (request.status === 500) {
+        this.notification.danger('An error occurred while syncing the project');
+      }
+    } catch (e) {
+      this.notification.danger(e);
+    }
+    this.loading = false;
+    this.store.findRecord('project', this.args.project.id);
+  }
```

Comment:
You can use a task from `ember-concurrency` for this. Among other nice functionality, the task will expose properties like `isRunning` to show a loading state.

Take the time to read about concurrency tasks and then refactor this action to a task.

Reply:
done here: https://github.com/adfinis/Outdated/pull/30/commits/e878e3f0785ab3ab24a4311052fd7e256f48b21c